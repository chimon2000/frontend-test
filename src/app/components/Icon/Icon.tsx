import * as React from "react";

export interface IconProps {
  children: string;
  className?: string;
}

export const Icon: React.FunctionComponent<IconProps> = ({
  children,
  className
}) => <i className={`icon material-icons ${className || ""}`}>{children}</i>;
