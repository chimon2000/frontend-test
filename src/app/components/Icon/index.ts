export { Icon as default, IconProps } from "./Icon";
