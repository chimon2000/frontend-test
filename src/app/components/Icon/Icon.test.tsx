import * as React from "react";
import { shallow } from "enzyme";

import { Icon } from "./Icon";

describe("<Icon />", () => {
  it("doesn't blow up", () => {
    const wrapper = shallow(<Icon>check</Icon>);

    expect(wrapper).toMatchSnapshot();
  });
});
