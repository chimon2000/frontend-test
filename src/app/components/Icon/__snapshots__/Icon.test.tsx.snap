// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> doesn't blow up 1`] = `
<i
  className="icon material-icons "
>
  check
</i>
`;
